src/backends/backend.js

```diff
@@ -70,8 +71,12 @@ class Backend {
   currentUser() {
     if (this.user) { return this.user; }
     const stored = this.authStore && this.authStore.retrieve();
-    if (stored) {
-      return Promise.resolve(this.implementation.setUser(stored)).then(() => stored);
+    if (stored && (stored.backend === this.backendName)) {
+      return this.implementation.restoreUser(stored).then((user) => {
+        // return confirmed/new user object instead of stored
+        this.authStore.store({...user, backend: this.backendName});
+        return {...user, backend: this.backendName};
```

Comment:
Nitpick: maybe store this object you're creating twice in a `newUser` constant.

Reply:
Done.